portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,13 +20,78 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import com.google.gson.Gson;
+
+/** Servlet that handles comments data.*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private static ArrayList<Comment> comments = new ArrayList<Comment>();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("text/html;");
+        response.getWriter().println(buildJsonFromList(comments));
+    }
+
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get new comment and add it to the list
+        String text = getParameter(request, "comment", "");
+        
+        // TODO Add properties to the comments (besides text)
+        Comment com = new Comment(text, "", "", "");        
+        comments.add(com);
+
+        response.setContentType("text/html;");
+        response.getWriter().println(buildJsonFromList(comments));
+
+        // Redirect back to the HTML page.
+        response.sendRedirect("/index.html");
+    }
+
+    // Converts a Comment object into a JSON
+    private String convertCommentToJson(Comment comment) {
+        String json = "{";
+        json += "\"text\": ";
+        json += "\"" + comment.getText() + "\"";
+        json += ", ";
+        json += "\"time\": ";
+        json += "\"" + comment.getTime() + "\"";
+        json += ", ";
+        json += "\"date\": ";
+        json += "\"" + comment.getDate() + "\"";
+        json += ", ";
+        json += "\"user\": ";
+        json += "\"" + comment.getUser() + "\"";
+        json += "}";
+        return json;
+    }
+
+    // Converts an array of Comment objects to a JSON array
+    private String buildJsonFromList(final ArrayList<Comment> list) {
+        String json = "{\"comments\": [";
+        for (Comment element : list) {
+            json += convertCommentToJson(element) + ", ";
+        }
+
+        // Delete the last ", " 
+        json = json.substring(0, json.length() - 2);
+        json += "]}";
+
+        return json;
+    }
```

Comment:
Gson gson = new Gson();
return gson.toJson(list);

Better practice is to avoid using string as a buffer object.
Gson allows user to stream result directly to output stream.